src/pngimage.cpp

```diff
@@ -189,14 +189,14 @@ void PngImage::printStructure(std::ostream& out, PrintStructureOption option, in
   chType[4] = 0;
 
   if (option == kpsBasic || option == kpsXMP || option == kpsIccProfile || option == kpsRecursive) {
-    constexpr auto xmpKey = "XML:com.adobe.xmp";
-    constexpr auto exifKey = "Raw profile type exif";
-    constexpr auto app1Key = "Raw profile type APP1";
-    constexpr auto iptcKey = "Raw profile type iptc";
-    constexpr auto iccKey = "icc";
-    constexpr auto softKey = "Software";
-    constexpr auto commKey = "Comment";
-    constexpr auto descKey = "Description";
+    auto xmpKey = upper("XML:com.adobe.xmp");
+    auto exifKey = upper("Raw profile type exif");
```

Comment:
Wonder if it's possible to do this is constexpr context.

Reply:
I removed the `constexpr` keyword from here for the same reason I was mentioning yesterday. At the end all these strings are used together with other string (`dataString`) which is obtained at runtime, and therefore we would not end up having computations at compile time. It might actually be decrimental to use `constexpre` here

If we cannot benefit from computations at compile time, I do not see why we should include the `constexpr` specifier. IMHO it decrements the readability of the codebase without a justification.

Thanks to your comment I realized that I should make these variables `const` because their values should not change.